### PR TITLE
Backbone representation

### PR DESCRIPTION
**Pull request by arose**
image

### CODE REVIEW

File: src/mol-math/linear-algebra/3d/vec3.ts [hunk 1]
```diff
@@ -474,7 +474,9 @@ namespace Vec3 {
 
     /** Computes the angle between 2 vectors, reports in radians. */
     export function angle(a: Vec3, b: Vec3) {
-        const theta = dot(a, b) / Math.sqrt(squaredMagnitude(a) * squaredMagnitude(b));
+        const denominator = Math.sqrt(squaredMagnitude(a) * squaredMagnitude(b));
+        if (denominator === 0) return Math.PI / 2;
```
Comment (dsehnal) [inline src/mol-math/linear-algebra/3d/vec3.ts:478]: Would ``denominator <= epsilon`` produce more stable results?

**Review by dsehnal — COMMENTED** [review]
Nice and definitely useful thanks to the ability to use impostors to render the backbone.

Would it also make sense to support spline interpolation so that it would look like cartoons without secondary structures? Or, alternatively, add "ignore secondary structure/backbone only" option for cartoon representation as well?

File: src/mol-math/linear-algebra/3d/vec3.ts (hunk 1 above)
Comment (arose) [reply src/mol-math/linear-algebra/3d/vec3.ts:478]: maybe, the current way is like three.js is doing it

**Comment by arose** [comment]
> 
> 
> Nice and definitely useful thanks to the ability to use impostors to render the backbone.
> 
> Would it also make sense to support spline interpolation so that it would look like cartoons without secondary structures? Or, alternatively, add "ignore secondary structure/backbone only" option for cartoon representation as well?

Isn't that what the putty representation with a uniform size theme does?

image

**Comment by dsehnal** [comment]
> 
> 
> > Would it also make sense to support spline interpolation so that it would look like cartoons without secondary structures? Or, alternatively, add "ignore secondary structure/backbone only" option for cartoon representation as well?
> 
> Isn't that what the putty representation with a uniform size theme does?
> 
> image

Right, I was thinking I was missing something.

File: src/mol-math/linear-algebra/3d/vec3.ts (hunk 1 above)
Comment (dsehnal) [reply src/mol-math/linear-algebra/3d/vec3.ts:478]: Ok I guess we could leave it as it is.

**Pull request merged**